### PR TITLE
Module newton

### PR DESCRIPTION
**Pull request by arthbous**
* Added a different file for param.py
* Fixed a bug in Newton method
* Made newton method work for vectorial computations see pnp_vector.py for the test
* Only issue left is with the expression in three dimensions in C++ in pnpmodule/expressions.py...the class has an error but I can not find where

### CODE REVIEW

**Comment by arthbous** [comment]
@maximilianmetti It works now. I put all the parameters in params.py.
I also added a "pnp_vector.py" file, that does the same thing as "pnp.py" but in vector form. It uses the same Newton solver that is now compatible with everything

File: params.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# Folders
+CLEAN = 'yes'
+DATA_DIR = "output/data_pnp/"
+IMG_DIR = "output/img_pnp/"
+PrintFig = 1                 # it prints every PrintFig newton's iterations
+PrintData = 1                # it prints every PrintData newton's iterations
+
+# Mesh
+N = [25, 5, 5]
+Lenghts = [10.0, 2.0, 2.0]
```
Comment (maximilianmetti) [inline params.py:15]: `Lengths`

File: pnp.py [hunk 2]
```diff
@@ -23,36 +31,44 @@
...
+files.CheckDir(IMG_DIR, params.CLEAN)
 
 # Create mesh and define function space
-Lx = 10.0
-Ly = 2.0
-Lz = 2.0
-P1 = Point(-Lx/2.0, -Ly/2.0, -Lz/2.0)
-P2 = Point(Lx/2.0, Ly/2.0, Lz/2.0)
-mesh = BoxMesh(P1, P2, 25, 5, 5)
+Lenghts = np.array(params.Lenghts)
```
Comment (maximilianmetti) [inline pnp.py:40]: `Lengths`

File: pnp_vector.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+parameters["allow_extrapolation"] = True
+
+# Check and create the directories
+DATA_DIR = params.DATA_DIR
+IMG_DIR = params.IMG_DIR
+files.CheckDir(DATA_DIR, params.CLEAN)
+files.CheckDir(IMG_DIR, params.CLEAN)
+
+# Create mesh and define function space
+Lenghts = np.array(params.Lenghts)
```
Comment (maximilianmetti) [inline pnp_vector.py:40]: `Lengths`

**Review by maximilianmetti — APPROVED** [review]
Looks good. We need to fix that variable name in the next PR

**Pull request merged**